atlasdb-config/src/main/java/com/palantir/atlasdb/factory/DefaultLockAndTimestampServiceFactory.java

```diff
@@ -0,0 +1,626 @@
+import com.palantir.lock.client.NamespacedCoalescingLeaderTimeGetter;
+import com.palantir.lock.client.NamespacedConjureLockWatchingService;
+import com.palantir.lock.client.ProfilingTimelockService;
+import com.palantir.lock.client.ReferenceTrackingWrapper;
+import com.palantir.lock.client.RemoteLockServiceAdapter;
+import com.palantir.lock.client.RemoteTimelockServiceAdapter;
+import com.palantir.lock.client.RequestBatchersFactory;
+import com.palantir.lock.client.TimeLockClient;
+import com.palantir.lock.client.metrics.TimeLockFeedbackBackgroundTask;
+import com.palantir.lock.impl.LegacyTimelockService;
+import com.palantir.lock.v2.NamespacedTimelockRpcClient;
+import com.palantir.lock.v2.TimelockRpcClient;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.lock.watch.LockWatchCache;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.timestamp.ManagedTimestampService;
+import com.palantir.timestamp.RemoteTimestampManagementAdapter;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampRange;
+import com.palantir.timestamp.TimestampService;
+import com.palantir.timestamp.TimestampStoreInvalidator;
+import com.palantir.util.OptionalResolver;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import javax.ws.rs.ClientErrorException;
+
+public final class DefaultLockAndTimestampServiceFactory implements LockAndTimestampServiceFactory {
+    private static final SafeLogger log = SafeLoggerFactory.get(DefaultLockAndTimestampServiceFactory.class);
+
+    private static final int ATTEMPTS_BEFORE_LOGGING_FAILURE_TO_READ_REMOTE_TIMESTAMP_SERVER_ID = 60;
+
+    private final MetricsManager metricsManager;
+    private final AtlasDbConfig config;
+    private final Refreshable<AtlasDbRuntimeConfig> runtimeConfig;
+    private final Consumer<Object> env;
+    private final Supplier<LockService> lock;
+    private final Supplier<ManagedTimestampService> time;
+    private final TimestampStoreInvalidator invalidator;
+    private final UserAgent userAgent;
+    private final Optional<LockDiagnosticComponents> lockDiagnosticComponents;
+    private final ReloadingFactory reloadingFactory;
+    private final Optional<TimeLockFeedbackBackgroundTask> timeLockFeedbackBackgroundTask;
+    private final Optional<TimeLockRequestBatcherProviders> timelockRequestBatcherProviders;
+    private final Set<Schema> schemas;
+
+    public DefaultLockAndTimestampServiceFactory(
+            MetricsManager metricsManager,
```

Comment:
all this stuff got turned into fields so the interface could have a no-arg method.